server/api/submission/submission.controller.js

```diff
@@ -226,26 +227,52 @@ exports.show = function(req, res) {
 
 // Creates a new Submission in the DB
 exports.create = function(req, res) {
-  req.body.userId = req.user._id;
+  if (req.user._id)
+    {
+      req.body.userId = req.user._id;
+    }
+
   saveSubmissionImage(req.files, req.body, (imagePaths)=>{
-    var submit = {
-      images : imagePaths,
-      submitter : req.body.userId,
-      authors : req.body.authors,
-      sectionEvent : req.body.eventId,
-      location : {
-        geo: {
+    req.body.coordinates[0] = Number(req.body.coordinates[0]);
+    req.body.coordinates[1] = Number(req.body.coordinates[1]);
+    if (!req.body.authors){
```

Comment:
Coordinates are not mandatory for submission, especially down the line instructors may not want to require them.